src/ActionList/Visuals.tsx

```diff
@@ -75,3 +77,56 @@ export const TrailingVisual: React.FC<React.PropsWithChildren<VisualProps>> = ({
     </Box>
   )
 }
+
+// VisualOrIndicator handles the positioning of indicators and determines whether to show a visual or an indicator.
+//
+// If we're showing an *inactive* or *loading* indicator and a leading visual has NOT been passed,
+// replace the trailing visual with the inactive indicator.
+//
+// This preserves the left alignment of item text.
+export const VisualOrIndicator: React.FC<
+  React.PropsWithChildren<{
+    inactiveText?: TooltipProps['text']
+    itemHasLeadingVisual: boolean
+    labelId?: string
+    loading?: boolean
+    position: 'leading' | 'trailing'
+  }>
+> = ({children, labelId, loading, inactiveText, itemHasLeadingVisual, position}) => {
+  const VisualComponent = position === 'leading' ? LeadingVisual : TrailingVisual
+
+  if (!loading && !inactiveText) return children
+
+  if (
+    (itemHasLeadingVisual && position === 'trailing') || // has a leading visual, and it's in the trailing position, or
+    (!itemHasLeadingVisual && position === 'leading') // it doesn't have a leading visual, and it's in the leading position
+  ) {
+    // => so we don't render the indicator here
+    return children
+  }
+
+  return inactiveText ? (
+    <Tooltip text={inactiveText}>
+      <Box
+        as="button"
+        sx={{
+          background: 'none',
+          color: 'inherit',
+          border: 'none',
+          padding: 0,
+          font: 'inherit',
+          cursor: 'pointer',
+        }}
+        aria-labelledby={labelId}
```

Comment:
I see we're using the `text` prop without defining the type, so it's defaulting to `description`. I'm thinking it might be better to switch this for the `label` type and having the `aria-labelledby` on the button here be `aria-describedby` instead. This would make the tooltip the accessible name, with the ActionList.Item inner text content being the description (e.g. "{inactiveText}, {ActionList.Item Text}").

Reply:
This sounds counter to the feedback I got during the same a11y office hours I mentioned in your previous comment.

Reply:
Thanks for the link :grin: I took a peek at the feedback, and I see that it's mentioned to use the `aria-describedby` version of the `Tooltip`, which I believe would still apply even if we did swap the values of `aria-describedby` and `labelledby`. Right now, the current usage of `aria-labelledby` means that we're relying on the text directly within `ActionList.Item` `<span>` as the accessible name, this causes redundancy since the previous element in the DOM carries the same accessible name.  

(e.g. current example
```html
<div>
  <span id="item-1--label">Action list item text contents</span>
  <button aria-labelledby="item-1--label" aria-describedby="tooltip-:r23:">
    <span><Icon /></span>
  </button>
  <div role="tooltip" id="tooltip-:r23:">Alert contents</div>
</div>
```
)

If a user is navigating through the DOM, "Action list item text contents" will be announced twice, once while on the `<span>` and second when on the `<button>`, which could be redundant. The second potential issue is that is that the purpose of the "alert" button is now secondary to the text within `ActionList.Item#item-(n)--label`, meaning that the text that describes the alert could be missed since it's only referred to with `aria-describedby`. 

I think ideally, we'd want the above example to announce something like this when a user is on the `<button>`: "`Accessible name`: Alert contents, `Accessible description`: Action list item text contents". 

(e.g. suggested example
```html
<div>
  <span id="item-1--label">Action list item text contents</span>
  <button aria-labelledby="tooltip-:r23:" aria-describedby="item-1--label">
    <span><Icon /></span>
  </button>
  <div role="tooltip" id="tooltip-:r23:">Alert contents</div>
</div>
```
)
The above example would remove the redundancy of "Action list item text contents", and ensures that the accessible name of the `<button>` is the actual purpose of the control. Screen reader users will be read the intent of the error message and then will be read what the error message refers to via the accessible description.

Tagging @primer/accessibility-design FR to make sure that this suggestion is aligned with the feedback given during office hours!

Reply:
I implemented this pattern and pushed a commit. I can easily revert if needed.

Reply:
Is the `Tooltip` still present? I'm trying to gain back context from my previous comments, but I was unable to find a story that had active tooltips in `ActionList` 🤔